### PR TITLE
Add SSO extension mode to device info

### PR DESCRIPTION
**Pull request by jasoncoolmax**
Expose SSO Extension mode in device info.

UTs added.

### CODE REVIEW

File: MSAL/src/public/MSALDefinitions.h [hunk 1]
```diff
@@ -154,6 +154,19 @@ typedef NS_ENUM(NSUInteger, MSALDeviceMode)
     MSALDeviceModeShared
 };
 
+/**
+ Device mode configured by the administrator
+ */
+typedef NS_ENUM(NSUInteger, MSALSSOExtensionMode)
```
Comment (oldalton) [inline MSAL/src/public/MSALDefinitions.h]: I don't think we need to expose device mode publicly in MSAL... is there any value from it to outside developers?

Comment (oldalton) [inline MSAL/src/public/MSALDefinitions.h]: Please remove this from public header.

File: MSAL/src/public/MSALDeviceInformation.h [hunk 2]
```diff
@@ -39,6 +39,11 @@ NS_ASSUME_NONNULL_BEGIN
 */
 @property (nonatomic, readonly) MSALDeviceMode deviceMode;
 
+/**
+ Additional device information
+*/
+- (NSDictionary *) extraDeviceInformation;
```
Comment (oldalton) [inline MSAL/src/public/MSALDeviceInformation.h]: nit: perhaps a readonly property would be more appropriate.

File: MSAL/src/MSALDeviceInformation.m [hunk 3]
```diff
@@ -65,6 +78,11 @@ - (NSString *)msalDeviceModeString
     }
 }
 
+- (void) initExtraDeviceInformation:(MSIDDeviceInfo *)deviceInfo
+{
+    [_extraDeviceInformation setValue:deviceInfo.ssoExtensionMode == MSIDSSOExtensionModeFull ? @"Yes" : @"No" forKey:MSAL_DEVICE_INFORMATION_SSO_EXTENSION_FULL_MODE_KEY];
```
Comment (oldalton) [inline MSAL/src/MSALDeviceInformation.m]: nit: any reason we're using @"Yes" and @"No" as strings? Can we just pass @NO and @YES? Or you would like to keep it as <NSString, NSString>?

Comment (jasoncoolmax) [reply MSAL/src/MSALDeviceInformation.m]: Just for readability as we did for the sso extension response. But I am fine if we just use NSNumber. Let me change it.

File: MSAL/src/public/MSALDefinitions.h (hunk 1 above)
Comment (jasoncoolmax) [reply MSAL/src/public/MSALDefinitions.h]: :+1: done

File: MSAL/src/public/MSALDeviceInformation.h (hunk 2 above)
Comment (jasoncoolmax) [reply MSAL/src/public/MSALDeviceInformation.h]: sure, changed :)

**Comment by jasoncoolmax** [comment]
UTs added.

File: MSAL/src/MSALDeviceInformation.m (hunk 3 above)
Comment (jasoncoolmax) [reply MSAL/src/MSALDeviceInformation.m]: added comment to the function above to emphasize that both key and value should be string :)

**Pull request merged**